proposals/dao/tip_vebalotc.ts

```diff
@@ -90,23 +90,25 @@ const validate: ValidateUpgradeFunc = async (addresses, oldContracts, contracts,
   const otcBuyerSigner = await getImpersonatedSigner(vebalOtcBuyer);
 
   // Check the proxy's state variables
-  expect(await contracts.veBoostManager.owner()).to.be.equal(addresses.vebalOtcHelper);
-  expect(await contracts.veBoostManager.votingEscrowDelegation()).to.be.equal(addresses.balancerVotingEscrowDelegation);
-  expect(await contracts.veBoostManager.balancerMinter()).to.be.equal(addresses.balancerMinter);
-  expect(await contracts.veBoostManager.gaugeController()).to.be.equal(addresses.balancerGaugeController);
-  expect(await contracts.veBoostManager.core()).to.be.equal(addresses.core);
+  expect(await contracts.balancerGaugeStaker.owner()).to.be.equal(addresses.vebalOtcHelper);
+  expect(await contracts.balancerGaugeStaker.votingEscrowDelegation()).to.be.equal(
+    addresses.balancerVotingEscrowDelegation
+  );
+  expect(await contracts.balancerGaugeStaker.balancerMinter()).to.be.equal(addresses.balancerMinter);
+  expect(await contracts.balancerGaugeStaker.gaugeController()).to.be.equal(addresses.balancerGaugeController);
+  expect(await contracts.balancerGaugeStaker.core()).to.be.equal(addresses.core);
 
   // Check veBAL OTC helper owner and proxy owner
   expect(await contracts.vebalOtcHelper.owner()).to.be.equal(vebalOtcBuyer);
-  expect(await contracts.veBoostManager.owner()).to.be.equal(contracts.vebalOtcHelper.address);
+  expect(await contracts.balancerGaugeStaker.owner()).to.be.equal(contracts.vebalOtcHelper.address);
   expect(
-    '0x' + (await ethers.provider.getStorageAt(addresses.veBoostManagerProxy, PROXY_ADMIN_STORAGE_SLOT)).slice(26)
+    '0x' + (await ethers.provider.getStorageAt(addresses.balancerGaugeStakerProxy, PROXY_ADMIN_STORAGE_SLOT)).slice(26)
   ).to.be.equal(vebalOtcBuyer.toLowerCase());
 
   // Check that OTC Buyer can transfer proxy ownership
-  await contracts.veBoostManagerProxy.connect(otcBuyerSigner).changeAdmin(addresses.feiDAOTimelock);
+  await contracts.balancerGaugeStakerProxy.connect(otcBuyerSigner).changeAdmin(addresses.feiDAOTimelock);
   expect(
-    '0x' + (await ethers.provider.getStorageAt(addresses.veBoostManagerProxy, PROXY_ADMIN_STORAGE_SLOT)).slice(26)
+    '0x' + (await ethers.provider.getStorageAt(addresses.balancerGaugeStakerProxy, PROXY_ADMIN_STORAGE_SLOT)).slice(26)
   ).to.be.equal(addresses.feiDAOTimelock.toLowerCase());
 };
```

Comment:
Lets check the implementation too.
`bytes32 internal constant _IMPLEMENTATION_SLOT = 0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc`

Reply:
Have added a check that the implementation address stored before the upgrade is the old implementation contract, and the one after the upgrade is the new implementation contract